js/repl/Repl.js

```diff
@@ -243,16 +255,25 @@ class Repl extends React.Component {
             options={options}
             placeholder="Write code here"
           />
-          <CodeMirrorPanel
-            className={styles.codeMirrorPanel}
-            code={state.compiled}
-            errorMessage={state.evalErrorMessage}
-            fileSize={state.meta.compiledSize}
-            info={state.debugEnvPreset ? state.envPresetDebugInfo : null}
-            options={options}
-            placeholder="Compiled output will be shown here"
-          />
-        </div>
+
+          <SandpackConsumer>
+            {(sandpack) => (
+              <CodeMirrorPanel
+                className={styles.codeMirrorPanel}
+                code={
+                  sandpack.managerState
+                  && sandpack.managerState.transpiledModules['/index.js:']
+                  && sandpack.managerState.transpiledModules['/index.js:'].source
+                  && sandpack.managerState.transpiledModules['/index.js:'].source.compiledCode}
+                errorMessage={state.evalErrorMessage}
+                fileSize={state.meta.compiledSize}
+                info={state.debugEnvPreset ? state.envPresetDebugInfo : null}
+                options={options}
+                placeholder="Compiled output will be shown here"
+              />
+            )}
+          </SandpackConsumer>
+        </SandpackProvider>
```

Comment:
That is really looking super nice. We can even conserve our styling, right?

Reply:
Yes! It's set up pretty flexible, in this case we can just give Babel the state of our bundler and that can be used in the UI. 😄